qiskit/providers/models/pulsedefaults.py

```diff
@@ -94,44 +98,335 @@ class Command(BaseModel):
-        cmd_def (list[Command]): Backend command definition.
+    """Description of default settings for Pulse systems. These are operations or settings that
+    may be good starting points for the Pulse user. The user may modify these defaults through
+    the provided methods to build a reference to custom operations, which may in turn be used
+    for building Schedules or converting circuits to Schedules.
     """
 
-    def __init__(self, qubit_freq_est, meas_freq_est, buffer,
-                 pulse_library, cmd_def, **kwargs):
-        self.qubit_freq_est = qubit_freq_est
-        self.meas_freq_est = meas_freq_est
+    def __init__(self,
+                 qubit_freq_est: List[float],
+                 meas_freq_est: List[float],
+                 buffer: int,
+                 pulse_library: List[PulseLibraryItem],
+                 cmd_def: List[Command],
+                 **kwargs: Dict[str, Any]):
+        """
+        Validate and reformat transport layer inputs to initialize.
+
+        Args:
+            qubit_freq_est: Estimated qubit frequencies in GHz.
+            meas_freq_est: Estimated measurement cavity frequencies in GHz.
+            buffer: Default buffer time (in units of dt) between pulses.
+            pulse_library: Pulse name and sample definitions.
+            cmd_def: Operation name and definition in terms of Commands.
+            **kwargs: Other attributes for the super class.
+        """
+        super().__init__(**kwargs)
+
         self.buffer = buffer
+        self.qubit_freq_est = [freq * 1e9 for freq in qubit_freq_est]
+        self.meas_freq_est = [freq * 1e9 for freq in meas_freq_est]
         self.pulse_library = pulse_library
         self.cmd_def = cmd_def
+        self.ops_def = InstructionScheduleMap(cmd_def, pulse_library)
+
+    def __str__(self):
+        qubit_freqs = [freq / 1e9 for freq in self.qubit_freq_est]
+        meas_freqs = [freq / 1e9 for freq in self.meas_freq_est]
+        qfreq = "Qubit Frequencies [GHz]\n{freqs}".format(freqs=qubit_freqs)
+        mfreq = "Measurement Frequencies [GHz]\n{freqs} ".format(freqs=meas_freqs)
+        return ("<{name}({ops}{qfreq}\n{mfreq})>"
+                "".format(name=self.__class__.__name__, ops=str(self.ops_def),
+                          qfreq=qfreq, mfreq=mfreq))
+
+    def build_cmd_def(self) -> 'InstructionScheduleMap':
+        """
+        Return the InstructionScheduleMap built for this PulseDefaults instance.
 
-        super().__init__(**kwargs)
+        Returns:
+            InstructionScheduleMap: `InstructionScheduleMap` instance generated from defaults
+        """
+        warnings.warn("This method is deprecated. Returning a InstructionScheduleMap instead. "
+                      "This can be accessed simply through the `ops_def` attribute of this "
+                      "PulseDefaults instance.", DeprecationWarning)
+        return self.ops_def
+
+
+class InstructionScheduleMap():
```

Comment:
This should continue to live in the `pulse` or `scheduler` module as it is a core object for `pulse`, not just related to transport.

Reply:
I thought we had already agreed this would be a fine place for it. It's related to transport insofar as it is a reformatted version of transport data. see: BackendConfiguration and BackendProperties